tests/unit/test_solvers.py

```diff
@@ -833,6 +829,165 @@ def test_kernel_herding_analytic_not_unique(self) -> None:
             solver_state.gramian_row_mean, expected_gramian_row_mean
         )
 
+    def test_kernel_herding_analytic_unique_weighted_data(self) -> None:
+        # pylint: disable=line-too-long
+        r"""
+        Test kernel herding on a weighted analytical example, with a unique coreset.
+
+        In this example, we have data of:
+
+        .. math::
+            x = \begin{pmatrix}
+                0.3 & 0.25 \\
+                0.4 & 0.2 \\
+                0.5 & 0.125
+            \end{pmatrix}
+
+        with weights:
+
+        .. math::
+            w = \begin{pmatrix}
+                0.8 \\
+                0.1 \\
+                0.1
+            \end{pmatrix}
+
+        and choose a ``length_scale`` of :math:`\frac{1}{\sqrt{2}}` to simplify
+        computations with the ``SquaredExponentialKernel``, in particular it becomes:
+
+        .. math::
+            k(x, y) = e^{-||x - y||^2}.
+
+        # TODO: Update from here
```

Comment:
Thank you so much for adding an example for weighted data as well, it is a very clear explanation. I believe you forgot to remove `# TODO`, Apart from that, I am happy to merge it :)